### ISSUE TITLE
Evaluation mode missing on validation and predict

### ISSUE THREAD

**Issue by ljtruong** [Seanny123/da-rnn#21]
This section in predict()
https://github.com/Seanny123/da-rnn/blob/8585806feee7726084bb2f733a3aaefada9c1f2f/main.py#L178-L181
should be changed to
```
with torch.no_grad():
  y_history = numpy_to_tvar(y_history)
  _, input_encoded = t_net.encoder(numpy_to_tvar(X))
  y_pred[y_slc] = t_net.decoder(input_encoded, y_history).cpu().data.numpy()
```
This is required to disable any gradient calculation during validation / prediction in training loop. From my understanding if not included, it wll update weights on each validation / prediction iteration.

This is original final_prediction output
![original](https://user-images.githubusercontent.com/8170076/90636260-b1c4d180-e26d-11ea-8ffa-c9f8f366bf47.PNG)

This is with grad turned off during validation / prediction.
![grad off](https://user-images.githubusercontent.com/8170076/90636186-98bc2080-e26d-11ea-8187-2462f096954d.PNG)
